services/net/extract-quotes/LLM/LLMService.cs

```diff
@@ -0,0 +1,604 @@
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.RateLimiting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using TNO.Core.Http;
+using TNO.Services.ExtractQuotes.Config;
+using TNO.Services.ExtractQuotes.CoreNLP.models;
+using TNO.Services.NLP.ExtractQuotes;
+namespace TNO.Services.ExtractQuotes.LLM;
+
+/// <summary>
+/// LLMService class, provides a way to extract quotes using a third-party LLM API compatible with OpenAI format.
+/// </summary>
+public class LLMService : ICoreNLPService
+{
+    #region Variables
+    private readonly IHttpRequestClient HttpClient;
+    private readonly ILogger<LLMService> Logger;
+    private readonly ExtractQuotesOptions Options;
+    private readonly TokenBucketRateLimiter _rateLimiter;
+    private int _failureCount = 0;
+    private int _primaryApiKeyIndex = 0;
+    private int _fallbackApiKeyIndex = 0;
+
+    /// <summary>
+    /// thread logging information
+    /// </summary>
+    /// <returns>string</returns>
+    private static string GetThreadInfo()
```

Comment:
Place method in the correct section.  Currently it's showing up in the variables.